src/token/src/canister/inspect.rs

```diff
@@ -79,7 +79,17 @@ fn inspect_message() {
             let state = state.borrow();
             let balances = &state.balances;
             if balances.0.contains_key(&caller) {
-                ic_cdk::api::call::accept_message();
+                if caller == state.stats.owner || m != "burn" {
```

Comment:
I can't make an actual github suggestion since this block contains code outside of the changes, but given the changes I would recommend changing the entire match arm:

```rust
        m if TRANSACTION_METHODS.contains(&m) => {
            // These methods requires that the caller have tokens.
            let state = CanisterState::get();
            let state = state.borrow();
            let balances = &state.balances;
            if !balances.0.contains_key(&caller) {
                ic_cdk::println!("Transaction method is not called by a stakeholder. Rejecting.");
                return;
            }

            // Anything but the `burn` method
            if caller == state.stats.owner || m != "burn" {
                ic_cdk::api::call::accept_message();
                return;
            }

            // It's the `burn` method and the caller isn't the owner.
            let from = ic_cdk::api::call::arg_data::<(Option<Principal>, Nat)>().0;
            if from.is_some() {
                ic_cdk::println!("Only the owner can burn other's tokens. Rejecting.");
                return;
            }

            ic_cdk::api::call::accept_message();
        }
```

This is less dense but easier to read